### DISCUSSION TITLE
Some modules are missing in 8.6.6 via npm

### DISCUSSION THREAD

**Discussion by fifzteen** [visgl/deck.gl#6586] [Q&A] [answered]
Hi!
When I install v8.6.6 by `npm install deck.gl@8.6.6`, the following modules are missing.

* aggregation-layers
* carto
* geo-layers
* layers

![npm_install_865_866](https://user-images.githubusercontent.com/25410805/151112026-29a6445a-8937-4fc7-9838-af6d2961df19.png)

I can use `@deck.gl/layers` or others by installing directly `npm install @deck.gl/layers`,
but this appends `@deck.gl/layers` to dependencies...

What could cause this?
I wish the same modules as in v8.6.5 to be installed.
Thanks.

#### steps to reproduce

* create empty directory
* move current to that
* `npm init`
* `npm install deck.gl@8.6.6`

#### environment

* macOS 11.6.2
* npm 8.1.4

#### related issues
#6493, #6585

**Comment by Pessimistress** [comment, marked as answer]
All submodules are listed as dependencies of the `deck.gl` package: https://unpkg.com/browse/deck.gl@8.6.6/package.json and they all exist on npm. This is a problem in your local configuration of npm.

If your npm is configured to use a private registry, the registry may not be synced with the upstream.

```bash
> npm config get registry
(Not https://registry.npmjs.org/)
```

You should find clues in the console log when you run npm install.

**Reply by fifzteen** [reply]
@Pessimistress 
Thank you for your reply!

First, I checked npm registry.

```sh
> npm cofig get registry
https://registry.npmjs.org/
```

OK. I don't use private registry.

Second, I told to npm not to use cache. (already removed package-lock.json and node_modules directory)

```sh
> npm install --cache /tmp/empty-cache --loglevel verbose deck.gl@8.6.6
```

I got same result. @deck.gl/layers and others are not in node_modules...

In verbose log, some errors about peer dependencies of @loaders.gl/core found.

```
npm WARN ERESOLVE overriding peer dependency
npm verb attempt to log [
  'ERESOLVE',
  'overriding peer dependency',
  {
    code: 'ERESOLVE',
    edge: {
      type: 'peer',
      name: '@loaders.gl/core',
      spec: '3.1.0',
      error: 'INVALID',
      from: [Object]
    },
    dep: {
      name: '@loaders.gl/core',
      version: '3.1.0',
      whileInstalling: [Object],
      location: 'node_modules/@loaders.gl/core',
      isWorkspace: false,
      dependents: [Array]
    },
    current: {
      name: '@loaders.gl/core',
      version: '3.1.6',
      location: 'node_modules/deck.gl/node_modules/@loaders.gl/core',
      isWorkspace: false,
      dependents: [Array]
    },
    peerConflict: {
      name: '@loaders.gl/core',
      version: '3.1.6',
      location: 'node_modules/deck.gl/node_modules/@loaders.gl/core',
      isWorkspace: false,
      dependents: [Array]
    },
    strictPeerDeps: false,
    force: false,
    isMine: false
  }
] crashed TypeError: Cannot destructure property 'name' of 'node' as it is undefined.
...
```

Here is all log.
[install.log](https://github.com/visgl/deck.gl/files/7946876/install.log)

---

So I add `--legacy-peer-deps` option, all modules are installed successfully!

```sh
> npm install --cache /tmp/empty-cache --loglevel verbose --legacy-peer-deps deck.gl@8.6.6
```

Here is log.
[install_leagcy-peer-deps.log](https://github.com/visgl/deck.gl/files/7946880/install_leagcy-peer-deps.log)

**Reply by Pessimistress** [reply]
Thank you for investigating. Opened an issue here: https://github.com/visgl/loaders.gl/issues/2070

**Reply by Pessimistress** [reply]
This is fixed upstream in loaders.gl 3.1.7.

**Reply by fifzteen** [reply]
I can install all moduels of v8.6.6 successfully!
Thanks for the quick fixing!
